### PR TITLE
Ball_node_fastener

### PR DESCRIPTION
**Pull request by obucklin**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [x] I added new functions/classes and made them available on a second-level import, e.g. `compas_timber.datastructures.Beam`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by obucklin** [comment]
@chenkasirer @papachap Ready for review

File: src/compas_timber/connections/ball_node.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+from compas_timber.elements import BallNodeFastener
+from compas_timber.utils import intersection_line_line_param
+
+from .joint import Joint
+from .solver import JointTopology
+
+
+class BallNodeJoint(Joint):
+    """Represents a ball node type joint which joins the ends of multiple beams,
```
Comment (chenkasirer) [inline src/compas_timber/connections/ball_node.py:11]: where does this sit in the whole discussion of "fastener is a result of the design" vs. "design must comply with pre-defined fastener"?

Comment (obucklin) [reply src/compas_timber/connections/ball_node.py:11]: 100% Fasteners respond to design.

**Review by chenkasirer — APPROVED** [review]
LGTM

**Pull request merged**